internal/metastore/kv/kv_catalog.go

```diff
@@ -306,45 +443,60 @@ func (kc *Catalog) DropCollection(ctx context.Context, collectionInfo *model.Col
 	return nil
 }
 
-func (kc *Catalog) DropPartition(ctx context.Context, collectionInfo *model.Collection, partitionID typeutil.UniqueID, ts typeutil.Timestamp) error {
-	collMeta := model.MarshalCollectionModel(collectionInfo)
-	k := path.Join(CollectionMetaPrefix, strconv.FormatInt(collectionInfo.CollectionID, 10))
-	v, err := proto.Marshal(collMeta)
-	if err != nil {
-		log.Error("drop partition marshal fail", zap.String("key", k), zap.Error(err))
-		return err
+func dropPartition(collMeta *pb.CollectionInfo, partitionID typeutil.UniqueID) {
+	if collMeta == nil {
+		return
+	}
+
+	{
+		loc := -1
+		for idx, partition := range collMeta.GetPartitions() {
```

Comment:
revert partitions addition from collection and ignore its' compatibility

Reply:
removed.